src/Service/Startup.cs

```diff
@@ -534,6 +544,42 @@ private void ConfigureAuthentication(IServiceCollection services, RuntimeConfigP
             }
         }
 
+        /// <summary>
+        /// Configure Application Insights Telemetry based on the loaded runtime configuration. If Application Insights
+        /// is enabled, we can track different events and metrics.
+        /// </summary>
+        /// <param name="runtimeConfigurationProvider">The provider used to load runtime configuration.</param>
+        /// <seealso cref="https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core#enable-application-insights-telemetry-collection"/>
+        private void ConfigureApplicationInsightsTelemetry(IApplicationBuilder app, RuntimeConfigProvider runtimeConfigurationProvider)
+        {
+            if (runtimeConfigurationProvider.TryGetConfig(out RuntimeConfig? runtimeConfig) && runtimeConfig.Runtime.Telemetry is not null
+                && runtimeConfig.Runtime.Telemetry.ApplicationInsights.Enabled)
+            {
+                AppInsightsOptions = runtimeConfig.Runtime.Telemetry.ApplicationInsights;
+
+                if (string.IsNullOrWhiteSpace(AppInsightsOptions.ConnectionString))
+                {
+                    _logger.LogError("Logs won't be sent to Application Insights as connection string is not set in the runtime config.");
+                    return;
+                }
+
+                AppTelemetryClient = app.ApplicationServices.GetRequiredService<TelemetryClient>();
+
+                // Update the TelemetryConfiguration object
+                TelemetryConfiguration telemetryConfiguration = AppTelemetryClient.TelemetryConfiguration;
+                telemetryConfiguration.ConnectionString = AppInsightsOptions.ConnectionString;
+
+                if (AppTelemetryClient is null)
+                {
+                    _logger.LogError("Telemetry client is not initialized.");
+                }
+
+                // Updating Startup Logger to Log from Startup Class.
+                ILoggerFactory? loggerFactory = Program.GetLoggerFactoryForLogLevel(MinimumLogLevel);
+                _logger = loggerFactory.CreateLogger<Startup>();
```

Comment:
I don't understand why this is needed. You're already using `_logger` prior to this. `_logger` is populated in the Startup class constructor with an instance of `ILogger<Startup> logger`

Reply:
the logger doesn't have telemetryClient info, so the logs from StartUp class won't go to the App Insights, that's why updating it.

Reply:
From offline discussion - Already emitted logs will not be sent to AppInsights after this update.